test/utility/test_util.h

```diff
@@ -0,0 +1,119 @@
+/*-
+ * Public Domain 2014-2016 MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include "wt_internal.h"			/* For __wt_XXX */
+
+#ifdef _WIN32
+	#define DIR_DELIM '\\'
+	#define RM_COMMAND "rd /s /q "
+#else
+	#define	DIR_DELIM '/'
+	#define RM_COMMAND "rm -rf "
+#endif
+
+#define	DEFAULT_DIR "WT_TEST"
+#define	MKDIR_COMMAND "mkdir "
+
+#ifdef _WIN32
+#include "windows_shim.h"
+#endif
+
+/* Generic option parsing structure shared by all test cases. */
+typedef struct {
+	char  *home;
+	char  *progname;
+	enum {	TABLE_COL=1,	/* Fixed-length column store */
+		TABLE_FIX=2,	/* Variable-length column store */
+		TABLE_ROW=3	/* Row-store */
+	} table_type;
+	bool	 preserve;
+	bool	 verbose;
+	uint64_t     nrecords;
+	uint64_t     nops;
+	uint64_t     nthreads;
+	uint64_t     n_append_threads;
+	uint64_t     n_read_threads;
+	uint64_t     n_write_threads;
```

Comment:
All of these should have comments.  Do we anticipate ever needing a thread indicator for creates, updates, deletes, truncates?  Having a specialized field for each, plus a general nthreads may be confusing later.